payjoin/src/send/v2/mod.rs

```diff
@@ -156,8 +156,9 @@ impl Sender {
             self.v1.min_fee_rate,
         )?;
         let hpke_ctx = HpkeContext::new(rs, &self.reply_key);
+        // TODO: Try to return the error as a proper HpkeError::PayloadTooLarge
         let body = encrypt_message_a(
-            body,
+            body.try_into().expect("Payload amount incorrect"),
```

Comment:
I failed to correctly apply a map_err() from a HpkeError to a CreateRequestError. But I'm not quite sure on how to accomplish this at the moment, I need to take a closer look at Error conversions

Reply:
I think keeping the pad_plaintext around for deduplicate error handling is justifiable, the comments on the original may have been overly ambitious. Perhaps start with the first comment and then take a stab at removing pad_plaintext altogether. @nothingmuch can you remember why we said the pad_plaintext function should not exist at all?

ref:

https://github.com/payjoin/rust-payjoin/pull/405/files#diff-2236ea3589c5398fd90a3ac7e8a077689c3b66adf9fdc3738b04ef41580b1c88R247-R249

Reply:
I have this currently, but ultimately It still feels like I moved the logic up to then simply duplicate the error handling.

I'm happy to scrap my current approach here and just go with using an array ref internally but I am perhaps also a little confused as you said about how the plan was to remove the pad plaintext as the plaintext is not really ever expected to be initially in the `PADDED_PLAINTEXT_#_LENGTH` as I first thought possible here to then just pass it down as an array but ultimately it does still need to be resized in my approach.

https://github.com/benalleng/rust-payjoin/blob/f00820bdb2735acf4305ac7259facd24cba1756a/payjoin/src/send/v2/mod.rs#L160-L169

Reply:
Would we be ok with making both a `pad_plaintext_a` and `pad_plaintext_b` function? if we really want to return fixed length arrays, there's not really a way to return two possible array lengths from a function in rust is there?

Reply:
One way to make this generic would be with a declarative macro `impl_pad_plaintext(a, PADDED_MSG_A_LENGTH)` that could produce pad_message_a and pad_message_b, but it might not really save much implementation detail and just introduce complexity.